tests/testthat/test-cache.R

```diff
@@ -53,116 +53,92 @@ test_that("clear_cache deletes the directory if no file is given", {
+  expect_warning(e3 <- load_cache(file = f1, path = d1))
+  expect_equal(ls(e2), character(0))
+  expect_equal(ls(e3), character(0))
 })
 
 # `save_cache`
 
 test_that("save_cache creates a directory if needed", {
-  with_mock(
-    `lintr::read_settings` = function(...) invisible(...),
-
-    e1 <- new.env(parent = emptyenv()),
-    d1 <- tempfile(pattern = "lintr_cache_"),
-    f1 <- "R/test.R",
+  e1 <- new.env(parent = emptyenv())
+  d1 <- tempfile(pattern = "lintr_cache_")
+  f1 <- "R/test.R"
 
-    expect_false(file.exists(d1)),
-    expect_false(file.exists(file.path(d1, fhash(f1)))),
+  expect_false(file.exists(d1))
+  expect_false(file.exists(file.path(d1, fhash(f1))))
 
-    save_cache(cache = e1, file = f1, path = d1),
+  save_cache(cache = e1, file = f1, path = d1)
 
-    expect_true(file.exists(d1)),
-    expect_true(file.info(d1)$isdir),
-    expect_true(file.exists(file.path(d1, fhash(f1))))
-  )
+  expect_true(file.exists(d1))
+  expect_true(file.info(d1)$isdir)
+  expect_true(file.exists(file.path(d1, fhash(f1))))
 })
 
 test_that("save_cache uses unambiguous cache file names", {
-  with_mock(
-    `lintr::read_settings` = function(...) invisible(...),
-
-    e1 <- new.env(parent = emptyenv()),
-    d1 <- tempfile(pattern = "lintr_cache_"),
-    f1 <- "R/test.R",
-    f2 <- "test.R",
+  e1 <- new.env(parent = emptyenv())
+  d1 <- tempfile(pattern = "lintr_cache_")
+  f1 <- "R/test.R"
+  f2 <- "test.R"
 
-    expect_false(file.exists(file.path(d1, fhash(f1)))),
-    expect_false(file.exists(file.path(d1, fhash(f2)))),
+  expect_false(file.exists(file.path(d1, fhash(f1))))
+  expect_false(file.exists(file.path(d1, fhash(f2))))
 
-    save_cache(cache = e1, file = f1, path = d1),
-    save_cache(cache = e1, file = f2, path = d1),
+  save_cache(cache = e1, file = f1, path = d1)
+  save_cache(cache = e1, file = f2, path = d1)
 
-    expect_true(fhash(f1) != fhash(f2)),
-    expect_true(file.exists(file.path(d1, fhash(f1)))),
-    expect_true(file.exists(file.path(d1, fhash(f2))))
-  )
+  expect_true(fhash(f1) != fhash(f2))
+  expect_true(file.exists(file.path(d1, fhash(f1))))
+  expect_true(file.exists(file.path(d1, fhash(f2))))
 })
 
 test_that("save_cache saves all non-hidden objects from the environment", {
-  with_mock(
-    `lintr::read_settings` = function(...) invisible(...),
-
-    e1 <- new.env(parent = emptyenv()),
-    e1$t1 <- 1,
-    e1$t2 <- 2,
+  e1 <- new.env(parent = emptyenv())
+  e1$t1 <- 1
+  e1$t2 <- 2
 
-    d1 <- tempfile(pattern = "lintr_cache_"),
-    f1 <- "R/test.R",
+  d1 <- tempfile(pattern = "lintr_cache_")
+  f1 <- "R/test.R"
 
-    save_cache(cache = e1, file = f1, path = d1),
+  save_cache(cache = e1, file = f1, path = d1)
 
-    e2 <- new.env(parent = emptyenv()),
-    load(file = file.path(d1, fhash(f1)), envir = e2),
+  e2 <- new.env(parent = emptyenv())
+  load(file = file.path(d1, fhash(f1)), envir = e2)
 
-    expect_equal(e1, e2)
-  )
+  expect_identical(as.list(e1), as.list(e2))
```

Comment:
If you use this approach, I think you need `all.names = TRUE` to verify the property implied by the test title.

OTOH I'm not sure what "non-hidden" means as I would have expected hidden to imply a variable starting with `.`, but that doesn't appear in the test.

Additionally, I'd use `expect_equal()` here because `expect_identical()` just implies exact numerical comparisons, which I don't think is important here.

Reply:
that's not my reading of the docs (?expect_identical):

> expect_identical() is the baseline; expect_equal() relaxes the test to ignore small numeric differences.

i.e., only use expect_equal to ignore numerical differences

Reply:
Hmmm. I guess it missing another sentence that says something like: "Since in most cases tiny numeric differences aren't important, we recommend using `expect_equal()`."

Reply:
:see_no_evil: our internal policy encourages `expect_identical()` now and we've basically switched all the `expect_equal()` tests already, except where `ignore*=` or `tolerance=` are needed. I have come to quite like it TBH. It encourages stronger tests.